src/services/tours/tours-sync-service.ts

```diff
@@ -0,0 +1,61 @@
+import LocalDatabaseService from '../local-database-service'
+import { Tour, UpdateOrCreateTour } from '../../types/tour'
+import { TourStatusType } from '../../enums/tour-status-type'
+import ToursService from './tours-service'
+import { SingleApiResponse } from '../../types/api'
+
+export default class ToursSyncService {
+  private localDatabaseService: LocalDatabaseService
+  private tourService: ToursService
+
+  constructor (token: string | undefined) {
+    this.localDatabaseService = new LocalDatabaseService()
+    this.tourService = new ToursService(token)
+  }
+
+  public async synchronizeTourData (): Promise<void> {
+    const localDatabaseService = new LocalDatabaseService()
+    const toursToSync: Tour[] = await localDatabaseService.getToursToSynchronize()
+    for (const tour of toursToSync) {
+      await this.handleTourSync(tour)
+    }
+  }
+
+  public async synchronizeCreatedTour (id: string | undefined, tour: UpdateOrCreateTour): Promise<SingleApiResponse<Tour>> {
+    // we can delete the local tour here as it gets freshly added (incl. correct id if successful)
+    await this.localDatabaseService.deleteTour(id)
+    return this.tourService.create(tour)
```

Comment:
If the `create` method fails, we would have lost the tour entry. Can't we flip it around, so we call `create` first and only delete the tour locally if the `create` was successful?

Reply:
updated due to refactoring of local db handling